### PR TITLE
add new tests for url changes

### PR DESCRIPTION
**Pull request by dualcnhq**
## Updates:

- add new test for search page URL changes for metadata search
- update file formatting

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,14 +4,15 @@
   "description": "",
   "main": "index.js",
   "scripts": {
-    "gh:test": "npx playwright test --config=./playwright.config.ts",
+    "gh:test": "npx playwright test --config=./playwright.config.ts --trace on",
```
Comment (dualcnhq) [inline package.json:7]: added param for tracing error logs - https://playwright.dev/docs/trace-viewer

File: tests/search/search-metadata-url.spec.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+test('search page check URL changes for sort filter menus click event', async ({ page }) => {
+	await page.goto('https://archive.org/search');
+	await page.waitForTimeout(3000);
+	await expect(page).toHaveURL(/search/);
+
+	const searchInput = page.locator('#text-input');
+	await searchInput.fill('cats');
+	await searchInput.press('Enter');
+
+	await page.waitForTimeout(3000);
```
Comment (iisa) [inline tests/search/search-metadata-url.spec.ts:13]: these timeouts are a doozy 😵

File: tests/search/search-metadata-url.spec.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import { expect } from '@playwright/test';
+import { test } from '../../fixtures';
+
+test('search page check URL changes for sort filter menus click event', async ({ page }) => {
+	await page.goto('https://archive.org/search');
+	await page.waitForTimeout(3000);
```
Comment (iisa) [inline tests/search/search-metadata-url.spec.ts:6]: we should not wait 3 seconds for a page to load without any search request. Q: I thought playwright can detect when page has loaded?

File: tests/search/search-metadata-url.spec.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
+import { expect } from '@playwright/test';
+import { test } from '../../fixtures';
+
+test('search page check URL changes for sort filter menus click event', async ({ page }) => {
```
Comment (iisa) [inline tests/search/search-metadata-url.spec.ts:4]:
```suggestion
test('Search Page: check URL changes when Sorting', async ({ page }) => {
```

**Review by iisa — APPROVED** [review]
looks good.
3 second timeouts make it look really long - Q: doesn't url change immediately on click? or something else? can we remove/reduce some? thanks!

**Comment by iisa** [comment]
looks good. bummed that we can't see the CI successfully run - this was the part of the promise of adding Playwright to test suite - that it builds CI 😅 - after this test, let's step back on writing new ones so we can fix CI. thanks!

File: package.json [hunk 5]
```diff
@@ -4,14 +4,15 @@
...
     "show:report": "npx playwright show-report",
     "test": "npx playwright test --config=./playwright.config.ts",
     "test:local": "BROWSERSTACK_LOCAL=true npx playwright test --config=./playwright-local.config.ts",
     "test:codegen": "npx playwright codegen",
     "test:chromium": "npx playwright test --project=chromium",
     "test:debug": "npx playwright test --debug",
-    "test:ui": "npx playwright test --ui"
+    "test:ui": "npx playwright test --ui",
+    "test:headed": "npx playwright test --headed"
```
Comment (iisa) [inline package.json:15]: q: what's the

**Comment by iisa** [comment]
merging now to bring into fold

**Comment by dualcnhq** [comment]
> looks good. 3 second timeouts make it look really long - Q: doesn't url change immediately on click? or something else? can we remove/reduce some? thanks!

it should be, but for some reason inside browserstack or the playwright browser, it takes time to take effect... but we still need to double check if this is just something to do with the browser configuration/test fixture configuration that also causes the browser timeouts in most of the tests while running it in parallel

**Pull request merged**